sdks/java/managed/src/main/java/org/apache/beam/sdk/managed/Managed.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.managed;
+
+import com.google.auto.value.AutoValue;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.schemas.utils.YamlUtils;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.annotations.VisibleForTesting;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.base.Preconditions;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.collect.ImmutableMap;
+
+public class Managed {
+
+  // TODO: Dynamically generate a list of supported transforms
+  public static final String ICEBERG = "iceberg";
+
+  public static Read read(String source) {
+
+    return new AutoValue_Managed_Read.Builder()
+        .setSource(
+            Preconditions.checkNotNull(
+                Read.TRANSFORMS.get(source.toLowerCase()),
+                "An unsupported source was specified: '%s'. Please specify one of the following sources: %s",
+                source,
+                Read.TRANSFORMS.keySet()))
+        .setSupportedIdentifiers(new ArrayList<>(Read.TRANSFORMS.values()))
+        .build();
+  }
+
+  @AutoValue
+  public abstract static class Read extends SchemaTransform {
+    public static final Map<String, String> TRANSFORMS =
+        ImmutableMap.<String, String>builder()
+            .put(ICEBERG, "beam:schematransform:org.apache.beam:iceberg_read:v1")
+            .build();
+
+    abstract String getSource();
+
+    abstract @Nullable String getConfig();
+
+    abstract @Nullable String getConfigUrl();
+
+    abstract List<String> getSupportedIdentifiers();
+
+    abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setSource(String sourceIdentifier);
+
+      abstract Builder setConfig(@Nullable String config);
```

Comment:
I suggest removing this method and making setConfigUrl() support local files so that anybody who has the Yaml file locally can use that.

Reply:
(it already should support it through LocalFileSystem so probably just test ?)

Reply:
This is included in [these lines](https://github.com/apache/beam/pull/30808/files#diff-666d799f826fb29116181c65747fcff142c9b2cef5cad804005b3f216ad46293R164-R168). There is also a test case for this [here](https://github.com/apache/beam/pull/30808/files#diff-5b10b7b441cda0a0625c14f543fae83ab78798d28a64fa121f1d1c532d331076R98). 

We use FileSystems, so hopefully this will support more than LocalFileSystem?

Reply:
Ack. It's fine since this is not in the public API.